spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml

```diff
@@ -26,7 +26,6 @@
 			<groupId>com.google.cloud.functions</groupId>
 			<artifactId>functions-framework-api</artifactId>
 			<version>${google.cloud.functions.api.version}</version>
-			<scope>provided</scope>
```

Comment:
What's the reason for this change? The jar should be provided by the functions runtime.

Reply:
The `spring-boot-maven-plugin` seems to be unable to package the JAR with the missing dependency. If I keep the scope `provided`, I get the following error:

```
[ERROR] Failed to execute goal org.springframework.boot:spring-boot-maven-plugin:2.3.0.BUILD-SNAPSHOT:repackage (repackage) on project function-sample-gcp: Execution repackage of goal org.springframework.boot:spring-boot-maven-plugin:2.3.0.BUILD-SNAPSHOT:repackage failed: A required class was missing while executing org.springframework.boot:spring-boot-maven-plugin:2.3.0.BUILD-SNAPSHOT:repackage: com/google/cloud/functions/HttpFunction
[ERROR] -----------------------------------------------------
[ERROR] realm =    plugin>org.springframework.boot:spring-boot-maven-plugin:2.3.0.BUILD-SNAPSHOT-231134182
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/home/dzou/.m2/repository/org/springframework/boot/spring-boot-maven-plugin/2.3.0.BUILD-SNAPSHOT/spring-boot-maven-plugin-2.3.0.BUILD-SNAPSHOT.jar
[ERROR] urls[1] = file:/home/dzou/IdeaProjects/spring-cloud-function/spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/target/spring-cloud-function-adapter-gcp-3.1.0.BUILD-SNAPSHOT.jar
[ERROR] urls[2] = file:/home/dzou/.m2/repository/com/google/code/gson/gson/2.8.6/gson-2.8.6.jar
...
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR] 
[ERROR] -----------------------------------------------------
[ERROR] : com.google.cloud.functions.HttpFunction
```